File: web-app/src/types/task.ts
export interface Task {
  id: string;
  description: string;
  completed: boolean;
  category: string;
  priority: string;
  created_at: string;
  updated_at: string;
}

export interface TaskCreate {
  description: string;
  category?: string;
  priority?: string;
}

export interface TaskUpdate {
  description?: string;
  completed?: boolean;
  category?: string;
  priority?: string;
}

File: api/models/task.py
from pydantic import BaseModel
from sqlmodel import SQLModel, Field
from typing import Optional


# SQLModel table that maps to the existing 'tasks' table in Neon DB
class Task(SQLModel, table=True):
    __tablename__ = "tasks"
    
    id: Optional[int] = Field(default=None, primary_key=True)
    description: str
    completed: bool = False
    category: str = "General"
    priority: str = "Medium"
    user_id: Optional[str] = None


# Pydantic models for API request/response
class TaskCreate(BaseModel):
    description: str
    completed: bool = False
    category: str = "General"
    priority: str = "Medium"


class TaskRead(BaseModel):
    id: int
    description: str
    completed: bool
    category: str
    priority: str
    user_id: Optional[str] = None


class TaskUpdate(BaseModel):
    description: Optional[str] = None
    completed: Optional[bool] = None
    category: Optional[str] = None
    priority: Optional[str] = None